cli/src/semgrep/git.py

```diff
@@ -296,52 +219,26 @@ def baseline_context(self) -> Iterator[None]:
 
         Raises CalledProcessError if any calls to git return non-zero exit code
         """
-        env = get_state().env
         # Reabort in case for some reason aborting in __init__ did not cause
         # semgrep to exit
-        self._abort_on_pending_changes()
-        self._abort_on_conflicting_untracked_paths(self.status)
-
-        logger.debug("Running git write-tree")
 
-        current_head = git_check_output(["git", "rev-parse", "HEAD"])
-        try:
-            merge_base_sha = self._get_git_merge_base()
-            logger.debug("Running git checkout for baseline context")
-            git_check_output(["git", "reset", "--hard", merge_base_sha])
-            logger.debug("Finished git checkout for baseline context")
-            yield
-        finally:
-            # Return to non-baseline state
-
-            # git checkout will fail if the checked-out index deletes all files in the repo
-            # In this case, we still want to continue without error.
-            # Note that we have no good way of detecting this issue without inspecting the checkout output
-            # message, which means we are fragile with respect to git version here.
-            logger.debug("Running git reset to return original context")
-            # nosem: use-git-check-output-helper
-            x = subprocess.run(
-                ["git", "reset", "--hard", current_head],
-                capture_output=True,
-                timeout=env.git_command_timeout,
-            )
-            logger.debug("Finished git reset to return original context")
-
-            if x.returncode != 0:
-                output = x.stderr.decode()
-                if (
-                    output
-                    and len(output) >= 2
-                    and "pathspec '.' did not match any file(s) known to git"
-                    in output.strip()
-                ):
-                    logger.debug(
-                        "Restoring git index failed due to total repository deletion; skipping checkout"
-                    )
-                else:
-                    raise Exception(
-                        f"Fatal error restoring Git state; please restore your repository state manually:\n{output}"
-                    )
+        cwd = Path.cwd()
+        git_root = get_git_root_path()
+        relative_path = cwd.relative_to(git_root)
+        with tempfile.TemporaryDirectory() as tmpdir:
+            try:
+                merge_base_sha = self._get_git_merge_base()
+                logger.debug("Running git checkout for baseline context")
```

Comment:
Can you clarify that what we're actually doing is checking out the baseline commit in a new directory?

Reply:
Yes, looks like this git worktree is a very nice command.
Can you explain why it's useful here and why it allowed you to remove so much code in this PR.

Reply:
Yea will do

Reply:
@aryx Just to clarify, we've removed a lot of code because we changed the approach. Previously, we refused to run baseline scans when running `git checkout` would cause the user to lose changes. This meant we had to check for dirty paths and similar. Now, we don't have to do any of those checks, because instead we're just making a git copy of the repo into a separate directory.

The comments on how git worktree works are really helpful though! Just wanted to clarify that it's doing something different from the removed code